### DISCUSSION TITLE
New Version Compile Issues!

### DISCUSSION THREAD

**Discussion by liuzhi136** [apache/incubator-mxnet#12244] [Q&A]
build/src/operator/rnn.o: In function `mxnet::op::RNNProp::CreateOperatorEx(mxnet::Context, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> >*, std::vector<int, std::allocator<int> >*) const':
rnn.cc:(.text+0x4ee71): undefined reference to `mxnet::Operator* mxnet::op::CreateOp<mshadow::gpu>(mxnet::op::RNNParam, int)'

I try to update mxnet on my computer to the current newest version. But I got a lot of errors like above.
I use cuda 8 and cudnn 5.1 during compiling. I clone the source code from the mxnet github website. Does any one meet the same issue like me ? and help out!

**Comment by vdantu** [comment]
@liuzhi136 : Just curious, would https://github.com/apache/incubator-mxnet/issues/7072 be solving the issue? 
@mxnet-label-bot : [C++, Question]

**Comment by liuzhi136** [comment]
@vdantu The issue you offered was help, which apart of mxnet's file can be compiled. But it still encountered the same problem for the rest part of mxnet.  It looks like these:
a - build/src/operator/convolution_v1_gpu.o
a - build/src/operator/correlation_gpu.o
a - build/src/operator/crop_gpu.o
a - build/src/operator/grid_generator_gpu.o
a - build/src/operator/identity_attach_KL_sparse_reg_gpu.o
a - build/src/operator/instance_norm_gpu.o
a - build/src/operator/l2_normalization_gpu.o
a - build/src/operator/leaky_relu_gpu.o
a - build/src/operator/loss_binary_op_gpu.o
a - build/src/operator/make_loss_gpu.o
a - build/src/operator/optimizer_op_gpu.o
a - build/src/operator/pad_gpu.o
a - build/src/operator/pooling_v1_gpu.o
a - build/src/operator/regression_output_gpu.o
a - build/src/operator/rnn_gpu.o
a - build/src/operator/roi_pooling_gpu.o
a - build/src/operator/sequence_last_gpu.o
a - build/src/operator/sequence_mask_gpu.o
a - build/src/operator/sequence_reverse_gpu.o
a - build/src/operator/slice_channel_gpu.o
a - build/src/operator/softmax_output_gpu.o
a - build/src/operator/spatial_transformer_gpu.o
a - build/src/operator/svm_output_gpu.o
a - build/src/operator/swapaxis_gpu.o

The rest still have the same problem.
build/src/operator/contrib/ctc_loss_gpu.o: In function `mxnet::op::CTCLossOp<mshadow::gpu>::~CTCLossOp()':
tmpxft_00000e38_00000000-4_ctc_loss.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEED2Ev[_ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEED5Ev]+0x50): undefined reference to `cudnnDestroyCTCLossDescriptor'
build/src/operator/contrib/ctc_loss_gpu.o: In function `mxnet::op::CTCLossOp<mshadow::gpu>::CTCLossOp(mxnet::op::CTCLossParam)':
tmpxft_00000e38_00000000-4_ctc_loss.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEEC2ENS0_12CTCLossParamE[_ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEEC5ENS0_12CTCLossParamE]+0x4a): undefined reference to `cudnnCreateCTCLossDescriptor'
tmpxft_00000e38_00000000-4_ctc_loss.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEEC2ENS0_12CTCLossParamE[_ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEEC5ENS0_12CTCLossParamE]+0x122): undefined reference to `cudnnSetCTCLossDescriptor'
build/src/operator/contrib/ctc_loss_gpu.o: In function `mxnet::op::CTCLossOp<mshadow::gpu>::cudnn_forward(mxnet::OpContext const&, mshadow::Stream<mshadow::gpu>*, mshadow::Tensor<mshadow::gpu, 3, float>, mshadow::Tensor<mshadow::gpu, 1, float>, mshadow::Tensor<mshadow::gpu, 3, float>, std::vector<int, std::allocator<int> >*, std::vector<int, std::allocator<int> >*, std::vector<int, std::allocator<int> >*, int, int, int, bool)':
tmpxft_00000e38_00000000-4_ctc_loss.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEE13cudnn_forwardERKNS_9OpContextEPNS2_6StreamIS3_EENS2_6TensorIS3_Li3EfEENSB_IS3_Li1EfEESC_PSt6vectorIiSaIiEESH_SH_iiib[_ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEE13cudnn_forwardERKNS_9OpContextEPNS2_6StreamIS3_EENS2_6TensorIS3_Li3EfEENSB_IS3_Li1EfEESC_PSt6vectorIiSaIiEESH_SH_iiib]+0x30a): undefined reference to `cudnnGetCTCLossWorkspaceSize'
tmpxft_00000e38_00000000-4_ctc_loss.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEE13cudnn_forwardERKNS_9OpContextEPNS2_6StreamIS3_EENS2_6TensorIS3_Li3EfEENSB_IS3_Li1EfEESC_PSt6vectorIiSaIiEESH_SH_iiib[_ZN5mxnet2op9CTCLossOpIN7mshadow3gpuEE13cudnn_forwardERKNS_9OpContextEPNS2_6StreamIS3_EENS2_6TensorIS3_Li3EfEENSB_IS3_Li1EfEESC_PSt6vectorIiSaIiEESH_SH_iiib]+0x752): undefined reference to `cudnnCTCLoss'
build/src/operator/nn/convolution_gpu.o: In function `MaxForwardAlgos(cudnnContext*)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._Z15MaxForwardAlgosP12cudnnContext[_Z15MaxForwardAlgosP12cudnnContext]+0x39): undefined reference to `cudnnGetConvolutionForwardAlgorithmMaxCount'
build/src/operator/nn/convolution_gpu.o: In function `MaxBackwardFilterAlgos(cudnnContext*)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._Z22MaxBackwardFilterAlgosP12cudnnContext[_Z22MaxBackwardFilterAlgosP12cudnnContext]+0x39): undefined reference to `cudnnGetConvolutionBackwardFilterAlgorithmMaxCount'
build/src/operator/nn/convolution_gpu.o: In function `MaxBackwardDataAlgos(cudnnContext*)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._Z20MaxBackwardDataAlgosP12cudnnContext[_Z20MaxBackwardDataAlgosP12cudnnContext]+0x39): undefined reference to `cudnnGetConvolutionBackwardDataAlgorithmMaxCount'
build/src/operator/nn/convolution_gpu.o: In function `mxnet::op::CuDNNConvolutionOp<float>::InitDescriptors(std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x28d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x35d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x42d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x4fc): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x5cb): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIfE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x69a): undefined reference to `cudnnSetConvolutionGroupCount'
build/src/operator/nn/convolution_gpu.o: In function `mxnet::op::CuDNNConvolutionOp<double>::InitDescriptors(std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x28d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x35d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x42d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x4fc): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x5cb): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_[_ZN5mxnet2op18CuDNNConvolutionOpIdE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS5_EES9_15cudnnDataType_tSA_]+0x69a): undefined reference to `cudnnSetConvolutionGroupCount'
build/src/operator/nn/convolution_gpu.o: In function `mxnet::op::CuDNNConvolutionOp<mshadow::half::half_t>::InitDescriptors(std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x28d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x35d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x42d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x4fc): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x5cb): undefined reference to `cudnnSetConvolutionGroupCount'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIN7mshadow4half6half_tEE15InitDescriptorsERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x69a): undefined reference to `cudnnSetConvolutionGroupCount'
build/src/operator/nn/convolution_gpu.o: In function `mxnet::op::CuDNNConvolutionOp<float>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)':
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x18d): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x270): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x34f): undefined reference to `cudnnSetConvolutionMathType'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0x69a): undefined reference to `cudnnGetConvolutionForwardAlgorithm_v7'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xc6b): undefined reference to `cudnnGetConvolutionBackwardFilterAlgorithm_v7'
tmpxft_0000210a_00000000-4_convolution.compute_61.cudafe1.cpp:(.text._ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_[_ZN5mxnet2op18CuDNNConvolutionOpIfE10SelectAlgoERKNS_10RunContextERKSt6vectorIN4nnvm6TShapeESaIS8_EESC_15cudnnDataType_tSD_]+0xf1c): undefined reference to `cudnnGetConvolutionBackwardDataAlgorithm_v7'
build/src/operator/nn/convolution_gpu.o: In function `mxnet::op::CuDNNConvolutionOp<double>::SelectAlgo(mxnet::RunContext const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, std::vector<nnvm::TShape, std::allocator<nnvm::TShape> > const&, cudnnDataType_t, cudnnDataType_t)':

**Comment by leleamol** [comment]
Hi @liuzhi136 ,
If you are building on Ubuntu, the build from source instructions on the links below would be helpful.

1. http://mxnet.incubator.apache.org/install/ubuntu_setup.html
2. http://mxnet.incubator.apache.org/install/build_from_source.html

The installation links mention Cuda 9.2 and CuDNN 7.14 versions.
I have tried these instructions on my setup and could build the mxnet without errors.

Please provide the build flags or config.mk file , it will help in root causing the issue.

**Comment by leleamol** [comment]
Hi @liuzhi136 ,

Please let us know if you were able to build the cpp-package after following documents mentioned in the last comment.
@mxnet-label-bot [pending-requestor-info]

**Comment by leleamol** [comment]
@mxnet-label-bot add [pending-requester-info]

**Comment by liuzhi136** [comment]
@leleamol No. It does not help. I tried these instructions. I don't no why this building process failed. I had successfully build mxnet from source before. Maybe it's some inner errors stemming from my ubuntu system.

**Comment by leleamol** [comment]
The issue doesn't seem to be related to C++ API. WIll remove the C++ label.

@liuzhi136 can you please upload the errors that you are getting when building from source. Also please share the config flags that you have used to build the mxnet.

**Comment by leleamol** [comment]
@mxnet-label-bot update [Pending Requester Info, Question, Build, Ubuntu]

**Comment by vdantu** [comment]
@liuzhi136 : Could you upload the config flags you are using? Are you still facing this issue? If you can attach the ```config.mk``` file here, we can try and recreate this issue.
